helix-view/src/editor.rs

```diff
@@ -1249,17 +1247,27 @@ impl Editor {
         }
     }
 
-    fn replace_document_in_view(&mut self, current_view: ViewId, doc_id: DocumentId) {
+    fn replace_document_in_view(
+        &mut self,
+        current_view: ViewId,
+        doc_id: DocumentId,
+        from_scratch: bool,
```

Comment:
I struggled to find a way to query if the current document was a scratch buffer.
However, the contexts in which this method is called have this knowledge.

Reply:
Detecting if it is a scratch buffer could be done with `doc!(self).path().is_none()` but not all scratch buffers are removed, only ones that aren't modified. We just want to protect against using `doc_mut!` with a DocumentId that doesn't exist:

```rust
// If swapping from a document, record the last view position
if let Some(old_doc) = self.documents.get_mut(&view.doc) {
    old_doc.last_view = view.offset;
}
```